src/test/java/uk/gov/hmcts/reform/sendletter/services/LetterServiceTest.java

```diff
@@ -80,12 +81,31 @@ public void setUp() {
 
     @Test
     public void should_return_letter_status_when_it_is_found_in_database() {
-        UUID letterId = UUID.randomUUID();
-        LetterStatus status = service.getStatus(letterId, "service-name");
+        LocalDateTime now = LocalDateTime.now();
+        Optional<LetterStatus> optional = Optional.of(new LetterStatus(
+            UUID.randomUUID(),
+            "some-message-id",
+            now,
+            now,
+            now
+        ));
+
+        given(letterRepository.getLetterStatus(optional.get().id, "service-name")).willReturn(optional);
+        LetterStatus status = service.getStatus(optional.get().id, "service-name");
+
+        assertThat(status).isEqualToComparingFieldByField(optional.get());
+    }
+
+    @Test
+    public void should_throw_letter_not_found_exception_when_not_in_database() {
+        UUID id = UUID.randomUUID();
+        given(letterRepository.getLetterStatus(id, "service-name")).willReturn(Optional.empty());
+
+        Throwable exception = catchThrowable(() -> service.getStatus(id, "service-name"));
 
-        assertThat(status)
-            .extracting("id")
-            .containsExactly(letterId);
+        assertThat(exception)
+            .isInstanceOf(LetterNotFoundException.class)
+            .hasMessage("Letter with '" + id.toString() + "' not found");
```

Comment:
Looks like the message is missing something :)
>Letter with 123 not found

Something to fix later I guess